File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
  int n = nums1.length;
        int m = nums2.length;

        int[] ans = new int[n];
        Arrays.fill(ans, -1);

        for (int i = 0; i < n; i++) {

            int a = nums1[i];

            for (int j = 0; j < m; j++) {

                if (nums2[j] == a) {

                    for (int p = j + 1; p < m; p++) {

                        if (nums2[p] > nums2[j]) {
                            ans[i] = nums2[p];
                            break;
                        }
                    }

                    break;
                }
            }
        }

        return ans;
    }
}

File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {

        int n=nums1.size();
        vector<int>ans(n,-1);

        for(int i=0;i<nums1.size();i++){
int p=-1;
            for(int  j=0;j<nums2.size();j++){
                if(nums2[j]==nums1[i]){
                    p=j;
                      for(int  k=p+1;k<nums2.size();k++){
                    if(nums2[k]>nums1[i]){
                        ans[i]=nums2[k];
                        break;
                    }
                }

                }
              
            }
        }
     return ans;   
    }
};